src/molecule.jl

```diff
@@ -0,0 +1,85 @@
+struct MoleculeName{M} end
+
+MoleculeName(mn::Symbol) = MoleculeName{mn}()
+MoleculeName(mn::String) = MoleculeName(Symbol(mn))
+
+macro MoleculeName_str(s)
+    :(MoleculeName{$(Expr(:quote, Symbol(s)))})
+end
+
+
+struct Atom
+  name::String
+  coordinates::NTuple{3, Real}
+end
+
+# attributes
+name(atom::Atom)       = atom.name
+coordinates(atom::Atom) = atom.coordinates
+
+Atom(name::String, x, y, z) = 
+  Atom(name, (x, y, z))
+
+# generate an atom at r = (0,0,0)
+Atom(name::String) = 
+  Atom(name, (0.0, 0.0, 0.0))
+
+struct Molecule
+  atoms::Vector{Atom}
+end
+
+atoms(molecule::Molecule) = 
+  molecule.atoms
+
+# empty molecule
+Molecule() = Molecule(Atom[])
+
+Molecule(a⃗::Atom...) = 
+  Molecule([a⃗...])
+
+Molecule(molecule::String) = Molecule(MoleculeName(molecule))
+Molecule(::MoleculeName{mn}) where {mn} = error("`molecule` not implemented for $mn")
+
+# parse atomic data from vectors of tuple/vectors
+Molecule(atoms::Vector{<:Tuple})  = 
+  Molecule([Atom(a[1], a[2:4]) for a in atoms])
+
+Molecule(atoms::Vector{<:Vector}) = 
+  Molecule(Tuple.(atoms))
+
+Base.length(molecule::Molecule) = 
+  length(atoms(molecule))
+
+Base.getindex(molecule::Molecule, i::Int) = 
+  atoms(molecule)[i] 
+
+Base.push!(molecule::Molecule, atom::Atom) = 
+  push!(molecule.atoms, atom)
+
+# parse data contained in `mol` to generate
+# an input string encoding for HF in Fermi.jl
+function parse_molecule(molecule::Molecule)
```

Comment:
@mtfishman This is an internal function just used to generate the `String` format to input to `Fermi.jl`

Reply:
I'm being a bit pedantic here, but when I hear `parse` I would think of it working in the opposite direction (parsing a string into an object, for example). Maybe we could call it `xyz_string` or something like that? Since it is outputting something similar to the XYZ file format: https://en.wikipedia.org/wiki/XYZ_file_format